includes/REST_API/Link_Controller.php

```diff
@@ -163,12 +163,22 @@ public function parse_link( $request ) {
 
 		// Do not request instagram.com, as it redirects to a login page.
 		// See https://github.com/GoogleForCreators/web-stories-wp/issues/10451.
-		$url_host = wp_parse_url( $url, PHP_URL_HOST );
-		if ( \is_string( $url_host ) && false !== strpos( $url_host, 'instagram.com' ) ) {
-			set_transient( $cache_key, wp_json_encode( $data ), $cache_ttl );
-			$response = $this->prepare_item_for_response( $data, $request );
+		$parse_url = wp_parse_url( $url );
```

Comment:
Maybe not in this PR but should this code be broken into a function for reuse outside of the `instagram.com` 

Possibly looping over an array of social sites as needed.

Reply:
I don't think that is in scope. But should be interesting for the future.